assembl/lib/piwik.py

```diff
@@ -0,0 +1,197 @@
+import requests
+from pyramid.settings import asbool
+
+def piwik_UsersManager_userExists(piwik_url, piwik_api_token, userLogin):
+    params = {
+        "module": "API",
+        "format": "JSON",
+        "token_auth": piwik_api_token
+    }
+    params["method"] = "UsersManager.userExists"
+    params["userLogin"] = userLogin # Piwik has two different fields for login and email, but a user can have the same value as login and email
+    result = requests.get(piwik_url, params=params, timeout=15)
+    if result.status_code != 200:
+        raise requests.ConnectionError()
+
+    content = result.json()
+    # print "piwik_UsersManager_userExists", content
+    if not content:
+        raise requests.ConnectionError()
+
+    user_already_exists = asbool(content.get("value", False))
+    return user_already_exists
+
+def piwik_UsersManager_getUserByEmail(piwik_url, piwik_api_token, userEmail):
+    params = {
+        "module": "API",
+        "format": "JSON",
+        "token_auth": piwik_api_token
+    }
+    params["method"] = "UsersManager.getUserByEmail"
+    params["userEmail"] = userEmail # Piwik has two different fields for login and email, but a user can have the same value as login and email
+    result = requests.get(piwik_url, params=params, timeout=15)
+    if result.status_code != 200:
+        raise requests.ConnectionError()
+
+    content = result.json() # returns something like [{"login":"aaa","email":"aaa@aaa.com"}] or {"result":"error","message":"L'utilisateur 'aaa@aaa.com' est inexistant."}
+    # print "piwik_UsersManager_getUserByEmail", content
+    if not content:
+        raise requests.ConnectionError()
+
+    if "result" in content and content["result"] == "error":
+        return False
+    elif not isinstance(content, list):
+        raise requests.ConnectionError()
+    else:
+        return content
+
+
+def piwik_UsersManager_addUser(piwik_url, piwik_api_token, userLogin, password, email, alias=''):
+    params = {
+        "module": "API",
+        "format": "JSON",
+        "token_auth": piwik_api_token
+    }
+    params["method"] = "UsersManager.addUser"
+    params["userLogin"] = userLogin # Piwik has two different fields for login and email, but a user can have the same value as login and email
+    params["password"] = password
+    params["email"] = email
+    if ( alias ):
```

Comment:
aren't parenthesis useless here?

Reply:
yes that is just a taste of aesthetic/readability. Do we use a rule that forces us to use the minimum amount of parentheses?

Reply:
My bad, I thought it was in PEP8 but it seems not. However, most linting tools wont like this (e.g. just tested it with pylint). When there is only one statement, if a comma "appears" inside, it will be seen as a tuple by python and so it will be always true whereas if you don't use parenthesis, your editor (or the interpretor) should complain on `if alias,:`.
Also, we generally don't use spaces inside parenthesis in python code.

Reply:
improvement done